### DISCUSSION TITLE
Concatenate fields in Grid view

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6834] [Help] [answered]
How can you concatenate first name and last name columns into one in the grid  using the client side grid

**Comment by luongnv6** [comment, marked as answer]
add new field in Row.cs


        [DisplayName("FullName"), Expression("T0.[FirstName] + ', ' + T0.[LastName]"), QuickSearch, MinSelectLevel(SelectLevel.Always)]
        public string FullName
        {
            get => fields.FullName[this];
            set => fields.FullName[this] = value;
        }


then add it to Column.cs

**Reply by rhemasystems** [reply]
But please I want to do this from client side using typescript because I have a Title  which is a string `enum `that I have to append to the concatenated full name, from the row its just an `enum `integer field stored in the database

**Reply by luongnv6** [reply]
try this:
```js
 protected getColumns() {
            var columns = super.getColumns();

            Q.first(columns, x => x.field == "FullName").format = (ctx) => {
                var item: abcRow = ctx.item;
                let title = Serenity.EnumFormatter.format(Serenity.EnumTypeRegistry.get(TitleEnumKey), Q.toId(item.Title));
                return `${title} ${item.FirstName} ${item.LastName}`;
            }

            return columns;
        }
```

**Reply by rhemasystems** [reply]
Thanks luongnv6
